python/cudf/cudf/core/column/string.py

```diff
@@ -5701,6 +5701,15 @@ def _binaryop(
             return NotImplemented
 
         if isinstance(other, (StringColumn, str, cudf.Scalar)):
+            if isinstance(other, cudf.Scalar) and other.dtype != "O":
+                if op in {"__eq__", "__lt__", "__le__", "__gt__", "__ge__"}:
+                    val = False
+                elif op == "__ne__":
+                    val = True
+                return column.full(len(self), val, dtype="bool").set_mask(
+                    self.mask
+                )
```

Comment:
```suggestion
                elif op == "__ne__":
                    val = True
                else:
                    raise TypeError(f"{op} between {self.dtype} and {other.dtype}") not supported")
                return column.full(len(self), val, dtype="bool").set_mask(
                    self.mask
                )
```

We might want to have an else block here to throw an error for rest of the ops like `+`, `-`, etc. Else we run into unbounded variable errors:

```python
In [2]: s = cudf.Series([1, 2, 1, 2], dtype='str')

In [5]: s + '1'
Out[5]: 
0    11
1    21
2    11
3    21
dtype: object

In [6]: s + 1
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
Cell In[6], line 1
----> 1 s + 1

File /nvme/0/pgali/envs/cudfdev/lib/python3.10/site-packages/cudf/core/mixins/mixin_factory.py:11, in _partialmethod.<locals>.wrapper(self, *args2, **kwargs2)
     10 def wrapper(self, *args2, **kwargs2):
---> 11     return method(self, *args1, *args2, **kwargs1, **kwargs2)

File /nvme/0/pgali/envs/cudfdev/lib/python3.10/site-packages/cudf/core/indexed_frame.py:3282, in IndexedFrame._binaryop(self, other, op, fill_value, can_reindex, *args, **kwargs)
   3278 if operands is NotImplemented:
   3279     return NotImplemented
   3281 return self._from_data(
-> 3282     ColumnAccessor(type(self)._colwise_binop(operands, op)),
   3283     index=out_index,
   3284 )

File /nvme/0/pgali/envs/cudfdev/lib/python3.10/contextlib.py:79, in ContextDecorator.__call__.<locals>.inner(*args, **kwds)
     76 @wraps(func)
     77 def inner(*args, **kwds):
     78     with self._recreate_cm():
---> 79         return func(*args, **kwds)

File /nvme/0/pgali/envs/cudfdev/lib/python3.10/site-packages/cudf/core/frame.py:1728, in Frame._colwise_binop(cls, operands, fn)
   1720         assert False, "At least one operand must be a column."
   1722 # TODO: Disable logical and binary operators between columns that
   1723 # are not numerical using the new binops mixin.
   1725 outcol = (
   1726     getattr(operator, fn)(right_column, left_column)
   1727     if reflect
-> 1728     else getattr(operator, fn)(left_column, right_column)
   1729 )
   1731 if output_mask is not None:
   1732     outcol = outcol.set_mask(output_mask)

File /nvme/0/pgali/envs/cudfdev/lib/python3.10/site-packages/cudf/core/mixins/mixin_factory.py:11, in _partialmethod.<locals>.wrapper(self, *args2, **kwargs2)
     10 def wrapper(self, *args2, **kwargs2):
---> 11     return method(self, *args1, *args2, **kwargs1, **kwargs2)

File /nvme/0/pgali/envs/cudfdev/lib/python3.10/site-packages/cudf/core/column/string.py:5709, in StringColumn._binaryop(self, other, op)
   5707     elif op == "__ne__":
   5708         val = True
-> 5709     return column.full(len(self), val, dtype="bool").set_mask(
   5710         self.mask
   5711     )
   5713 if op == "__add__":
   5714     if isinstance(other, cudf.Scalar):

UnboundLocalError: local variable 'val' referenced before assignment
```

Reply:
Adjusted this